document-store/src/main/java/org/hypertrace/core/documentstore/postgres/utils/PostgresUtils.java

```diff
@@ -94,11 +94,8 @@ public static String prepareCast(String field, Type type) {
       return String.format(fmt, field, type);
     } else if (type.equals(Type.BOOLEAN)) {
       return String.format(fmt, field, type);
-    } else {
-      // default is string
-      // Note: As we are using field accessor pattern, we are casting String too.
-      // See more details on method : prepareParsedNonCompositeFilter
-      return String.format(fmt, field, type);
+    } else /* default is string */ {
```

Comment:
As mentioned in the description of the PR, reverting back to the data accessor pattern `->>`